### PR TITLE
Collision detection

### PR DESCRIPTION
**Pull request by EmceeEscher**
Got collision detection working (just bounding box, no meshes). Seems to correctly identify when two cars in intersection collide, but doesn't do anything about it. Only checks cars for `if_in_beyond_intersec` returns true.

### CODE REVIEW

File: src/lane_manager.cpp [hunk 1]
```diff
@@ -28,13 +28,60 @@ bool LaneManager::update(float ms)
...
+
+			if (first_car->check_collision(second_bb.bottom_left)
+				|| first_car->check_collision(second_bb.bottom_right)
+				|| first_car->check_collision(second_bb.top_right)
+				|| first_car->check_collision(second_bb.top_left)) {
+					collision_occurring = true;
+					printf("first_car getting hit!\n");
+					// TODO: if this happens, check triangles in mesh, then give car new velocity
+				}
+			else if (second_car->check_collision(first_bb.bottom_left)
```
Comment (EmceeEscher) [inline src/lane_manager.cpp:70]: not sure if this `else` should be here or not. Usually both will be true, but occasionally only the second will be.

Comment (theochit) [reply src/lane_manager.cpp:70]: Hmm that's what I was thinking too, but if there's any corner case where the second is the only one that's true, then let's definitely keep it.

**Review by theochit — APPROVED** [review]
Looks like a really good start. The only thing I'm a little unsure about is how the same collision is detected many times. Obviously this will be lower when the cars separate from the impact, but it still feels like once there is an impact, it should be announced once and the cars enter a new state (something like is_crashing, rather than in_intersection). 

Anyway, I haven't thought it through much so if you don't think that's a problem then go ahead and merge.

**Comment by EmceeEscher** [comment]
Re: detecting the same collision multiple times

I think we will probably have to have some `is_crashing`state (and probably make it so that a car can't be `is_crashing` and `is_in_beyond_intersec` at the same time), but I don't know if we want cars that are in the process of crashing to also be able to cause more crashes? If we do, then we have to keep checking them anyway... My inclination is to have the crash actually do something and then we can see if it's going to be a problem.

**Pull request merged**